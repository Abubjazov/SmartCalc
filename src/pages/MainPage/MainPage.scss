@import '../../variables';

.main-page {
	background: $white;

	.container {
		padding: 11vh 0 1vh 0;
		min-height: 100vh;
		max-width: calc(100vw - 2vw);
		display: grid;
		grid-template-columns: auto-fit, 100px;
		grid-template-rows: 150px 25px;
		row-gap: 10px;
		justify-items: center;

		@media (max-width: 900px) {
			padding: 6vh 0 0 0;
		}

		a {
			border-radius: 20px;
			padding-top: 11px;
			width: 100px;
			height: 40px;
			text-align: center;
			font-size: 18px;
			line-height: 18px;
			background-color: $main-color;
			color: $white;
			text-decoration: none;
			transition: all 0.1s linear;
		}
	}
}
