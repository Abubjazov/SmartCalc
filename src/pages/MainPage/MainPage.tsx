import { NavLink } from 'react-router-dom'
import { Helmet } from 'react-helmet'

import { useTypedSelector } from '../../hooks/useTypedSelector'

import './MainPage.scss'

export const MainPage = (): JSX.Element => {
	const { token } = useTypedSelector(state => state.auth)
	return (
		<main className='main-page'>
			<Helmet>
				<title>Главная</title>
			</Helmet>
			{/* <PageMarker pageName='main' /> */}

			<div className='container'>
				<div className='calc-info'>
					<h2>Smart Calc - простой онлайн калькулятор.</h2>
					<hr />
					<br />
					Smart C - простое онлайн приложение на основе React, Redux и
					Typescript.
					<br />
					<br />
					<h3>Логин и пароль для Demo:</h3>
					<br />
					login: test@test.com <br /> password: password
					<br />
					<br />
					<h3>На данный момент приложение имеет 3 реализованных раздела:</h3>
					<br />
					-Главная страница - первое что видит пользователь.
					<br />
					<br />
					- Страница ввода логина и пароля - где пользователь уже
					зарегистрированный в системе может ввести свой логин и пароль и
					попасть на страницу "Калькулятор".
					<br />
					<br />- Страница калькулятора - где пользователь может ввести данные и
					получить результат расчёта.
				</div>
				{token ? (
					<NavLink to='/smartcalc/calc'>Калькулятор</NavLink>
				) : (
					<NavLink to='/smartcalc/login'>Авторизация</NavLink>
				)}
			</div>
		</main>
	)
}
