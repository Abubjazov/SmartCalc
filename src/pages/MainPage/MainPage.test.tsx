import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store'

import { MainPage } from './MainPage'

const mockStore = configureStore()

describe('Component: MainPage', () => {
	test('should render MainPage with "Войти"', () => {
		const store = mockStore({
			auth: {
				token: null,
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<BrowserRouter>
					<MainPage />
				</BrowserRouter>
			</Provider>
		)

		expect(asFragment()).toMatchSnapshot()
	})

	test('should render MainPage with "Калькулятор"', () => {
		const store = mockStore({
			auth: {
				token: 'testtoken',
			},
		})

		const { asFragment } = render(
			<Provider store={store}>
				<BrowserRouter>
					<MainPage />
				</BrowserRouter>
			</Provider>
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
