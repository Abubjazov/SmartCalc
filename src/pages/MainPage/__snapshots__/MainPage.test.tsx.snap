// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MainPage should render MainPage with "Войти" 1`] = `
<DocumentFragment>
  <main
    class="main-page"
  >
    <div
      class="container"
    >
      <div
        class="calc-info"
      >
        <h2>
          SMART CALC V.2
        </h2>
        <h3>
          Онлайн калькулятор
        </h3>
        <p>
          SPA на основе React, Redux и Typescript
        </p>
        <p>
          Репозиторий проекта:
        </p>
        <a
          href="https://github.com/Abubjazov/SmartCalc"
          rel="noopener noreferrer"
          target="_blank"
        >
          github.com/Abubjazov/SmartCalc
        </a>
        <p>
          Документация:
        </p>
        <a
          href="https://github.com/Abubjazov/SmartCalc/blob/main/README.md"
          rel="noopener noreferrer"
          target="_blank"
        >
          README.md
        </a>
        <p>
          Demo сервер:
        </p>
        <a
          href="https://abubjazov.github.io/SmartCalc/"
          rel="noopener noreferrer"
          target="_blank"
        >
          github.io/SmartCalc/
        </a>
        <p>
          Стэк технологий:
        </p>
        <h3>
          Typescript, React, ReactRouter, ReactHelmet, Redux, ReduxThunk, Axios, Nanoid, Bcryptjs, SASS
        </h3>
        <p
          class="calc-info-clarification"
        >
          Все права защищены! © Abubjazov 2022 (
          <a
            href="https://github.com/Abubjazov"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://github.com/Abubjazov
          </a>
          )
        </p>
      </div>
      <a
        class="main-page-a"
        href="/smartcalc/login"
      >
        Войти
      </a>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Component: MainPage should render MainPage with "Калькулятор" 1`] = `
<DocumentFragment>
  <main
    class="main-page"
  >
    <div
      class="container"
    >
      <div
        class="calc-info"
      >
        <h2>
          SMART CALC V.2
        </h2>
        <h3>
          Онлайн калькулятор
        </h3>
        <p>
          SPA на основе React, Redux и Typescript
        </p>
        <p>
          Репозиторий проекта:
        </p>
        <a
          href="https://github.com/Abubjazov/SmartCalc"
          rel="noopener noreferrer"
          target="_blank"
        >
          github.com/Abubjazov/SmartCalc
        </a>
        <p>
          Документация:
        </p>
        <a
          href="https://github.com/Abubjazov/SmartCalc/blob/main/README.md"
          rel="noopener noreferrer"
          target="_blank"
        >
          README.md
        </a>
        <p>
          Demo сервер:
        </p>
        <a
          href="https://abubjazov.github.io/SmartCalc/"
          rel="noopener noreferrer"
          target="_blank"
        >
          github.io/SmartCalc/
        </a>
        <p>
          Стэк технологий:
        </p>
        <h3>
          Typescript, React, ReactRouter, ReactHelmet, Redux, ReduxThunk, Axios, Nanoid, Bcryptjs, SASS
        </h3>
        <p
          class="calc-info-clarification"
        >
          Все права защищены! © Abubjazov 2022 (
          <a
            href="https://github.com/Abubjazov"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://github.com/Abubjazov
          </a>
          )
        </p>
      </div>
      <a
        class="main-page-a"
        href="/smartcalc/calc"
      >
        Калькулятор
      </a>
    </div>
  </main>
</DocumentFragment>
`;
